### PR TITLE
feat(PE-4468): use turbo-sdk for uploading signed data items

### PR DESCRIPTION
**Pull request by dtfiedler**
## Details

Uses beta build of `turbo-sdk` to publish data items to turbo.

Note: 
- testing with `pnp` is really tough - not sure if my env is just not setup correctly or what, so switched to using `node-modules` making it much easier to test/validate the outputs of the SDK
- added a comment about `skipLibCheck` - this will require looking further into `arbundles` and `arweave` deps of the `turbo-sdk` that create some of the issues mentioned below. while ideally we have type safety, it's not uncommon to have to set this flag to avoid type checking on third-party libraries

### CODE REVIEW

**Comment by fedellen** [comment]
Pulled this down to try building and can't quite get it to resolve yet. Will dig further later.

getting this verbose output: 

```sh
~/ar-io/ardrive-core-js PE-4472-use-turbo-sdk *4 !1                                                                        11:35:35 AM
> yarn      
➤ YN0000: ┌ Resolution step
➤ YN0013: │ @ardrive/turbo-sdk@file:../ardrive-turbo-sdk/lib#../ardrive-turbo-sdk/lib::hash=30694d&locator=ardrive-core-js%40workspace
➤ YN0002: │ arbundles@npm:0.6.19 doesn't provide @types/node (pa2aa9), requested by ts-node
➤ YN0002: │ arbundles@npm:0.6.19 doesn't provide typescript (p5f1ab), requested by ts-node
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 0s 290ms
➤ YN0000: ┌ Fetch step
➤ YN0013: │ @ardrive/turbo-sdk@file:../ardrive-turbo-sdk/lib#../ardrive-turbo-sdk/lib::hash=30694d&locator=ardrive-core-js%40workspace
➤ YN0019: │ @ardrive-turbo-sdk-file-a27ab613d1-5052f2d6b4.zip appears to be unused - removing
➤ YN0000: └ Completed in 0s 262ms
➤ YN0000: ┌ Link step
➤ YN0000: │ ESM support for PnP uses the experimental loader API and is therefore experimental
➤ YN0007: │ @ardrive/turbo-sdk@file:../ardrive-turbo-sdk/lib#../ardrive-turbo-sdk/lib::hash=30694d&locator=ardrive-core-js%40workspace%3A. must be built because it never has been before or the last one failed
➤ YN0009: │ @ardrive/turbo-sdk@file:../ardrive-turbo-sdk/lib#../ardrive-turbo-sdk/lib::hash=30694d&locator=ardrive-core-js%40workspace%3A. couldn't be built successfully (exit code 127, logs can be found here: /private/var/folders/mp/_57qn31d1w36kvw1rlytfypr0000gn/T/xfs-84cab883/build.log)
➤ YN0000: └ Completed
➤ YN0000: Failed with errors in 0s 757ms

~/ar-io/ardrive-core-js PE-4472-use-turbo-sdk *4 !4 ?1                                                                     11:35:37 AM
> yarn build
.yarn/cache/arbundles-npm-0.9.9-1af4846417-b10da8dfec.zip/node_modules/arbundles/build/node/esm/src/signing/chains/injectedSolanaSigner.d.ts:3:49 - error TS2307: Cannot find module '@solana/wallet-adapter-base' or its corresponding type declarations.

3 import type { MessageSignerWalletAdapter } from "@solana/wallet-adapter-base";
                                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.yarn/cache/arweave-npm-1.11.8-628114908d-cab953c8ea.zip/node_modules/arweave/node/transactions.d.ts:1:22 - error TS6053: File '/Users/derek/ar-io/ardrive-core-js/.yarn/cache/arweave-npm-1.11.8-628114908d-cab953c8ea.zip/src/modules.d.ts' not found.

1 /// <reference path="../../../src/modules.d.ts" />
                       ~~~~~~~~~~~~~~~~~~~~~~~~~

.yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/common/http.d.ts:1:23 - error TS1452: 'resolution-mode' assertions are only supported when `moduleResolution` is `node16` or `nodenext`.

1 /// <reference types="node" resolution-mode="require"/>
                        ~~~~

.yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/common/http.d.ts:2:23 - error TS1452: 'resolution-mode' assertions are only supported when `moduleResolution` is `node16` or `nodenext`.

2 /// <reference types="node" resolution-mode="require"/>
                        ~~~~

.yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/common/http.d.ts:3:23 - error TS1452: 'resolution-mode' assertions are only supported when `moduleResolution` is `node16` or `nodenext`.

3 /// <reference types="node" resolution-mode="require"/>
                        ~~~~

.yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/common/http.d.ts:21:29 - error TS2307: Cannot find module 'retry-axios' or its corresponding type declarations.

21 import { RetryConfig } from 'retry-axios';
                               ~~~~~~~~~~~~~

.yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/node/signer.d.ts:1:23 - error TS1452: 'resolution-mode' assertions are only supported when `moduleResolution` is `node16` or `nodenext`.

1 /// <reference types="node" resolution-mode="require"/>
                        ~~~~

.yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/types/turbo.d.ts:1:23 - error TS1452: 'resolution-mode' assertions are only supported when `moduleResolution` is `node16` or `nodenext`.

1 /// <reference types="node" resolution-mode="require"/>
                        ~~~~

.yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/types/turbo.d.ts:2:23 - error TS1452: 'resolution-mode' assertions are only supported when `moduleResolution` is `node16` or `nodenext`.

2 /// <reference types="node" resolution-mode="require"/>
                        ~~~~

.yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/types/turbo.d.ts:3:23 - error TS1452: 'resolution-mode' assertions are only supported when `moduleResolution` is `node16` or `nodenext`.

3 /// <reference types="node" resolution-mode="require"/>
                        ~~~~

.yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/types/turbo.d.ts:22:29 - error TS2307: Cannot find module 'retry-axios' or its corresponding type declarations.

22 import { RetryConfig } from 'retry-axios';
                               ~~~~~~~~~~~~~

src/ardrive.ts:222:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSMoveFileResult'.

222    dataCaches,
       ~~~~~~~~~~

src/ardrive.ts:223:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSMoveFileResult'.

223    fastFinalityIndexes,
       ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:240:6 - error TS2322: Type '{ type: "file"; metadataTxId: TransactionID; dataTxId: TransactionID; entityId: EntityID; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

240      dataCaches,
         ~~~~~~~~~~

src/ardrive.ts:306:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSMoveFileResult & WithFileKey'.

306    dataCaches,
       ~~~~~~~~~~

src/ardrive.ts:307:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSMoveFileResult & WithFileKey'.

307    fastFinalityIndexes,
       ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:325:6 - error TS2322: Type '{ type: "file"; metadataTxId: TransactionID; dataTxId: TransactionID; entityId: EntityID; key: EntityKey; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

325      dataCaches,
         ~~~~~~~~~~

src/ardrive.ts:390:25 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSWriteResult'.

390   const { metaDataTxId, dataCaches, fastFinalityIndexes, metaDataTxReward } = await this.arFsDao.movePublicFolder(
                            ~~~~~~~~~~

src/ardrive.ts:390:37 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSWriteResult'.

390   const { metaDataTxId, dataCaches, fastFinalityIndexes, metaDataTxReward } = await this.arFsDao.movePublicFolder(
                                        ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:406:6 - error TS2322: Type '{ type: "folder"; metadataTxId: TransactionID; entityId: EntityID; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

406      dataCaches,
         ~~~~~~~~~~

src/ardrive.ts:483:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSWriteResult & WithDriveKey'.

483    dataCaches,
       ~~~~~~~~~~

src/ardrive.ts:484:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSWriteResult & WithDriveKey'.

484    fastFinalityIndexes,
       ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:501:6 - error TS2322: Type '{ type: "folder"; metadataTxId: TransactionID; entityId: EntityID; key: EntityKey; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

501      dataCaches,
         ~~~~~~~~~~

src/ardrive.ts:644:4 - error TS2339: Property 'dataCaches' does not exist on type 'FolderResult'.

644    dataCaches,
       ~~~~~~~~~~

src/ardrive.ts:645:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'FolderResult'.

645    fastFinalityIndexes
       ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:655:5 - error TS2345: Argument of type '{ type: "folder"; entityId: EntityID; metadataTxId: TransactionID; key: EntityKey | undefined; bundledIn: TransactionID | undefined; entityName: string; sourceUri: string | undefined; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to parameter of type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

655     dataCaches,
        ~~~~~~~~~~

src/ardrive.ts:676:4 - error TS2339: Property 'dataCaches' does not exist on type 'FileResult'.

676    dataCaches,
       ~~~~~~~~~~

src/ardrive.ts:677:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'FileResult'.

677    fastFinalityIndexes
       ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:688:5 - error TS2345: Argument of type '{ type: "file"; entityName: string; entityId: EntityID; dataTxId: TransactionID; metadataTxId: TransactionID; bundledIn: TransactionID | undefined; sourceUri: string | undefined; key: EntityKey | undefined; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to parameter of type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

688     dataCaches,
        ~~~~~~~~~~

src/ardrive.ts:1061:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSCreateFolderResult'.

1061    dataCaches,
        ~~~~~~~~~~

src/ardrive.ts:1062:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSCreateFolderResult'.

1062    fastFinalityIndexes
        ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:1077:6 - error TS2322: Type '{ type: "folder"; metadataTxId: TransactionID; entityId: EntityID; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

1077      dataCaches,
          ~~~~~~~~~~

src/ardrive.ts:1127:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSCreateFolderResult'.

1127    dataCaches,
        ~~~~~~~~~~

src/ardrive.ts:1128:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSCreateFolderResult'.

1128    fastFinalityIndexes
        ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:1144:6 - error TS2322: Type '{ type: "folder"; metadataTxId: TransactionID; entityId: EntityID; key: EntityKey; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

1144      dataCaches,
          ~~~~~~~~~~

src/ardrive.ts:1182:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSCreateBundledDriveResult | ArFSCreateDriveResult'.

1182    dataCaches,
        ~~~~~~~~~~

src/ardrive.ts:1183:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSCreateBundledDriveResult | ArFSCreateDriveResult'.

1183    fastFinalityIndexes
        ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:1192:6 - error TS2322: Type '{ type: "drive"; metadataTxId: TransactionID; entityId: EntityID; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

1192      dataCaches,
          ~~~~~~~~~~

src/ardrive.ts:1199:6 - error TS2322: Type '{ type: "folder"; metadataTxId: TransactionID; entityId: EntityID; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

1199      dataCaches,
          ~~~~~~~~~~

src/ardrive.ts:1520:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSRenameFileResult'.

1520    dataCaches,
        ~~~~~~~~~~

src/ardrive.ts:1521:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSRenameFileResult'.

1521    fastFinalityIndexes,
        ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:1536:6 - error TS2322: Type '{ type: "file"; entityId: EntityID; metadataTxId: TransactionID; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

1536      dataCaches,
          ~~~~~~~~~~

src/ardrive.ts:1582:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSRenameFileResult & WithFileKey'.

1582    dataCaches,
        ~~~~~~~~~~

src/ardrive.ts:1583:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSRenameFileResult & WithFileKey'.

1583    fastFinalityIndexes,
        ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:1600:6 - error TS2322: Type '{ type: "file"; entityId: EntityID; key: EntityKey; metadataTxId: TransactionID; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

1600      dataCaches,
          ~~~~~~~~~~

src/ardrive.ts:1642:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSRenameFolderResult'.

1642    dataCaches,
        ~~~~~~~~~~

src/ardrive.ts:1643:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSRenameFolderResult'.

1643    fastFinalityIndexes,
        ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:1658:6 - error TS2322: Type '{ type: "folder"; entityId: EntityID; metadataTxId: TransactionID; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

1658      dataCaches,
          ~~~~~~~~~~

src/ardrive.ts:1704:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSRenameFolderResult & WithDriveKey'.

1704    dataCaches,
        ~~~~~~~~~~

src/ardrive.ts:1705:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSRenameFolderResult & WithDriveKey'.

1705    fastFinalityIndexes,
        ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:1722:6 - error TS2322: Type '{ type: "folder"; entityId: EntityID; metadataTxId: TransactionID; key: EntityKey; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

1722      dataCaches,
          ~~~~~~~~~~

src/ardrive.ts:1762:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSRenameDriveResult'.

1762    dataCaches,
        ~~~~~~~~~~

src/ardrive.ts:1763:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSRenameDriveResult'.

1763    fastFinalityIndexes,
        ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:1778:6 - error TS2322: Type '{ type: "drive"; entityId: EntityID; metadataTxId: TransactionID; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

1778      dataCaches,
          ~~~~~~~~~~

src/ardrive.ts:1819:4 - error TS2339: Property 'dataCaches' does not exist on type 'ArFSRenameDriveResult & WithDriveKey'.

1819    dataCaches,
        ~~~~~~~~~~

src/ardrive.ts:1820:4 - error TS2339: Property 'fastFinalityIndexes' does not exist on type 'ArFSRenameDriveResult & WithDriveKey'.

1820    fastFinalityIndexes,
        ~~~~~~~~~~~~~~~~~~~

src/ardrive.ts:1837:6 - error TS2322: Type '{ type: "drive"; entityId: EntityID; key: EntityKey; metadataTxId: TransactionID; entityName: string; dataCaches: any; fastFinalityIndexes: any; }' is not assignable to type 'ArFSEntityData'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSEntityData'.

1837      dataCaches,
          ~~~~~~~~~~

src/ardrive_factory.ts:106:3 - error TS2345: Argument of type 'TurboUnauthenticatedClient' is not assignable to parameter of type 'Turbo'.
  Type 'TurboUnauthenticatedClient' is missing the following properties from type 'Turbo': isDryRun, turbo, sendDataItem

106   turboSettings === undefined
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~
107    ? defaultTurbo
    ~~~~~~~~~~~~~~~~~
... 
115      // TODO: add private key
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
116      })
    ~~~~~~~

src/arfs/arfs_entity_result_factory.ts:14:10 - error TS2305: Module '"./turbo"' has no exported member 'TurboCachesResponse'.

14 import { TurboCachesResponse } from './turbo';
            ~~~~~~~~~~~~~~~~~~~

src/arfs/arfsdao.ts:183:45 - error TS2307: Cannot find module '@ardrive/turbo-sdk' or its corresponding type declarations.

183 import { TurboUploadDataItemResponse } from '@ardrive/turbo-sdk';
                                                ~~~~~~~~~~~~~~~~~~~~

src/arfs/arfsdao.ts:307:4 - error TS2322: Type '{ folderId: EntityID; metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSCreateFolderResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSCreateFolderResult'.

307    dataCaches,
       ~~~~~~~~~~

src/arfs/arfsdao.ts:329:4 - error TS2322: Type '{ folderId: EntityID; metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSCreateFolderResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSCreateFolderResult'.

329    dataCaches,
       ~~~~~~~~~~

src/arfs/arfsdao.ts:454:4 - error TS2322: Type '{ driveId: EntityID; metaDataTxId: TransactionID; rootFolderId: EntityID; rootFolderTxId: TransactionID; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; bundleTxId: TransactionID; }' is not assignable to type 'ArFSCreateDriveToTurboResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSCreateDriveToTurboResult'.

454    dataCaches,
       ~~~~~~~~~~

src/arfs/arfsdao.ts:551:4 - error TS2322: Type '{ id: TransactionID; }' is not assignable to type '{ id: TransactionID; } & Pick<TurboUploadDataItemResponse, "dataCaches" | "fastFinalityIndexes">'.
  Type '{ id: TransactionID; }' is missing the following properties from type 'Pick<TurboUploadDataItemResponse, "dataCaches" | "fastFinalityIndexes">': dataCaches, fastFinalityIndexes

551    return { id: TxID(metaDataTx.id) };
       ~~~~~~

src/arfs/arfsdao.ts:586:4 - error TS2322: Type '{ dataTxId: TransactionID; metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSMoveFileResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSMoveFileResult'.

586    dataCaches,
       ~~~~~~~~~~

src/arfs/arfsdao.ts:619:4 - error TS2322: Type '{ dataTxId: TransactionID; metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; fileKey: EntityKey; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSMovePrivateFileResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSMovePrivateFileResult'.

619    dataCaches,
       ~~~~~~~~~~

src/arfs/arfsdao.ts:643:76 - error TS2322: Type '{ metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSWriteResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSWriteResult'.

643   return { metaDataTxId: id, metaDataTxReward: metaDataBaseReward?.reward, dataCaches, fastFinalityIndexes };
                                                                               ~~~~~~~~~~

src/arfs/arfsdao.ts:673:4 - error TS2322: Type '{ metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; driveKey: EntityKey; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSMovePrivateFolderResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSMovePrivateFolderResult'.

673    dataCaches,
       ~~~~~~~~~~

src/arfs/arfsdao.ts:918:56 - error TS2339: Property 'turboUrl' does not exist on type 'Turbo'.

918    console.error(`\nUploading to Turbo at ${this.turbo.turboUrl}...\n`);
                                                           ~~~~~~~~

src/arfs/arfsdao.ts:1071:4 - error TS2322: Type '{ entityId: EntityID; folderTxId: TransactionID; driveKey: EntityKey | undefined; entityName: string; sourceUri: string; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'FolderResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'FolderResult'.

1071    dataCaches,
        ~~~~~~~~~~

src/arfs/arfsdao.ts:1114:4 - error TS2322: Type '{ sourceUri: string | undefined; entityName: string; fileDataTxId: TransactionID; entityId: EntityID; metaDataTxId: TransactionID; fileKey: EntityKey | undefined; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; bundledIn: TransactionID; }' is not assignable to type 'FileResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'FileResult'.

1114    dataCaches,
        ~~~~~~~~~~

src/arfs/arfsdao.ts:2024:4 - error TS2322: Type '{ entityId: EntityID; metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSRenameFileResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSRenameFileResult'.

2024    dataCaches,
        ~~~~~~~~~~

src/arfs/arfsdao.ts:2067:4 - error TS2322: Type '{ entityId: EntityID; metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; fileKey: EntityKey; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSRenamePrivateFileResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSRenamePrivateFileResult'.

2067    dataCaches,
        ~~~~~~~~~~

src/arfs/arfsdao.ts:2098:4 - error TS2322: Type '{ entityId: EntityID; metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSRenameFolderResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSRenameFolderResult'.

2098    dataCaches,
        ~~~~~~~~~~

src/arfs/arfsdao.ts:2131:4 - error TS2322: Type '{ entityId: EntityID; metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; driveKey: EntityKey; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSRenamePrivateFolderResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSRenamePrivateFolderResult'.

2131    dataCaches,
        ~~~~~~~~~~

src/arfs/arfsdao.ts:2160:4 - error TS2322: Type '{ entityId: EntityID; metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSRenameDriveResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSRenameDriveResult'.

2160    dataCaches,
        ~~~~~~~~~~

src/arfs/arfsdao.ts:2191:4 - error TS2322: Type '{ entityId: EntityID; metaDataTxId: TransactionID; metaDataTxReward: Winston | undefined; driveKey: EntityKey; dataCaches: TurboUploadDataItemResponse; fastFinalityIndexes: TurboUploadDataItemResponse; }' is not assignable to type 'ArFSRenamePrivateDriveResult'.
  Object literal may only specify known properties, and 'dataCaches' does not exist in type 'ArFSRenamePrivateDriveResult'.

2191    dataCaches,
        ~~~~~~~~~~

src/arfs/turbo.ts:6:45 - error TS2307: Cannot find module '@ardrive/turbo-sdk' or its corresponding type declarations.

6 import { TurboUploadDataItemResponse } from '@ardrive/turbo-sdk';
                                              ~~~~~~~~~~~~~~~~~~~~

src/types/ardrive_types.ts:19:10 - error TS2305: Module '"../arfs/turbo"' has no exported member 'TurboCachesResponse'.

19 import { TurboCachesResponse } from '../arfs/turbo';
            ~~~~~~~~~~~~~~~~~~~


Found 79 errors in 11 files.

Errors  Files
     1  .yarn/cache/arbundles-npm-0.9.9-1af4846417-b10da8dfec.zip/node_modules/arbundles/build/node/esm/src/signing/chains/injectedSolanaSigner.d.ts:3
     1  .yarn/cache/arweave-npm-1.11.8-628114908d-cab953c8ea.zip/node_modules/arweave/node/transactions.d.ts:1
     4  .yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/common/http.d.ts:1
     1  .yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/node/signer.d.ts:1
     4  .yarn/unplugged/@ardrive-turbo-sdk-file-e0c5cbb6a3/node_modules/@ardrive/turbo-sdk/types/turbo.d.ts:1
    46  src/ardrive.ts:222
     1  src/ardrive_factory.ts:106
     1  src/arfs/arfs_entity_result_factory.ts:14
    18  src/arfs/arfsdao.ts:183
     1  src/arfs/turbo.ts:6
     1  src/types/ardrive_types.ts:19
```

File: tsconfig.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
 		"strict": true,
 		"esModuleInterop": true,
 		"resolveJsonModule": true,
+		"skipLibCheck": true,
```
Comment (dtfiedler) [inline tsconfig.json:11]: there are `type` errors in `arbundles` and `arweave` dependencies that cause issues when pulling in the `turbo-sdk`. we can ignore these errors while we work to resolve (either create PR's in those libraries or find a better way to handle them in the turbo-sdk).

File: src/ardrive_factory.ts [hunk 2]
```diff
@@ -93,8 +96,12 @@ export function arDriveFactory({
...
 		undefined,
 		turboSettings === undefined
-			? undefined
-			: new Turbo({ turboUrl: turboSettings.turboUrl ?? turboProdUrl, isDryRun: dryRun })
+			? defaultTurbo
+			: new Turbo({
+				turboUploadUrl: turboSettings.turboUploadUrl,
+				turboPaymentUrl: turboSettings.turboPaymentUrl,
+				isDryRun: turboSettings.isDryRun
+			})
```
Comment (fedellen) [inline src/ardrive_factory.ts:104]: I think this should provide each default here with turboSettings having these as optional. As it is right now, CLI only provides an upload service URL via `--turbo-url` parameter

File: src/arfs/turbo.ts [hunk 3]
```diff
@@ -1,64 +1,50 @@
...
-
-	private get dataItemEndpoint(): string {
-		return `${this.turboUrl.href}v1/tx`;
+		this.turbo = TurboFactory.unauthenticated({
+			uploadServiceConfig: {
+				url: turboUploadUrl.toString()
+			},
+			paymentServiceConfig: {
+				url: turboPaymentUrl.toString()
+			}
```
Comment (fedellen) [inline src/arfs/turbo.ts:30]: I had to make this change to get the pathing to line up. On the SDK side we add `/v1` but toString() on URL type gives `https://upload.ardrive.io/`, so we go to `https://upload.ardrive.io//v1` and get a 404. `origin` returns `https://upload.ardrive.io` and that works. A better route might be to use the URL type on the SDK side (if we like that type) or start making some custom url logic like trimming any trailing `/` on we're compatible with both inputs 

```suggestion
			uploadServiceConfig: {
				url: turboUploadUrl.origin
			},
			paymentServiceConfig: {
				url: turboPaymentUrl.origin
			}
```

File: src/ardrive_factory.ts (hunk 2 above)
Comment (fedellen) [reply src/ardrive_factory.ts:104]: CLI could maintain backwards compatibility and still target payment `dev` env here by using the upload service url just replacing `upload` and `payment`

Comment (dtfiedler) [reply src/ardrive_factory.ts]: e2f54fec415a1024aa538428cc33a40d4db53edf

**Review by fedellen — APPROVED** [review]

**Review by fedellen — APPROVED** [review]

**Pull request merged**